server/schedule/placement/fit.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package placement
+
+import (
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server/core"
+)
+
+// RegionFit is the result of fitting a region's peers to rule list.
+// All peers are divided into corresponding rules according to the matching
+// rules, and the remaining Peers are placed in the OrphanPeers list.
+type RegionFit struct {
```

Comment:
Is that possible one region can match multiple rules?

Reply:
Yes.

Reply:
So, the region will be put into the first `RuleFit` it matched?

Reply:
Not regions. Peers will be put into `RuleFit` it matches.
Not first. The fitting process is performed according to the rules. Each rule selects at most N matched peers, the rest will left to other Rules.

Reply:
I see. Thanks.